contrib/node/src/python/pants/contrib/node/subsystems/eslint_distribution.py

```diff
@@ -0,0 +1,133 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import hashlib
+import logging
+import os
+import shutil
+
+from pants.subsystem.subsystem import Subsystem
+from pants.util.dirutil import safe_mkdir, safe_rmtree
+
+logger = logging.getLogger(__name__)
+
+class ESLintDistribution(object):
+  """Represents a self-bootstrapping ESLint distribution."""
+
+  ESLINT_DISTRIBUTION_MODULE = 'synthetic-install-eslint-distribution-module'
+
+  class Factory(Subsystem):
+    options_scope = 'eslint-distribution'
+
+    @classmethod
+    def register_options(cls, register):
+      super(ESLintDistribution.Factory, cls).register_options(register)
+      register('--setupdir', advanced=True, fingerprint=True,
+               help='Find the package.json under this dir for installing eslint and plugins.')
+      register('--supportdir', advanced=True, default='bin/eslint',
+               help='Find the ESLint distribution under this dir.')
+      register('--eslint-config', advanced=True, fingerprint=True,
+               help='The path to the global eslint configuration file specifying all the rules')
+      register('--eslint-ignore', advanced=True, fingerprint=True,
+               help='The path to the global eslint ignore path')
+    def create(cls):
+      # NB: create is an instance method used to globally scope this subsystem.
+      options = cls.get_options()
+      global_options = cls.global_instance().get_options()
+      pants_bootstrapdir = global_options.pants_bootstrapdir
+      return ESLintDistribution(options.setupdir, options.supportdir, options.eslint_config,
```

Comment:
I will suggest to use NodeDistribution instead of creating a new Distribution. Add a TODO will be suffice.

Reply:
Done